src/reducers/user.reducer.ts

```diff
@@ -16,39 +16,48 @@ export const userInitialState: UserState = {
   email: '',
   rootAdmin: false,
   isRegistered: false,
-  roles: [],
+  userRoles: [],
   isLoggedIn: false,
 };
 
 export function userReducer(state = userInitialState, action: any): UserState {
+  let reduced = state;
```

Comment:
I think this may be a bit of an anti-pattern here. Each case in a reducer is meant to be a totally self-contained thing, with an emphasis on as few mutations as possible (preferably none). So sharing any kind of variable between them seems like it could lead to trouble if someone decided to get "clever" with that `reduced` variable. It's best to keep every case completely isolated so that it's easy to reason about them independently.